packages/multichain/src/caip25Permission.ts

```diff
@@ -283,17 +292,26 @@ function removeScope(
     newOptionalScopes.length !==
     Object.keys(caip25CaveatValue.optionalScopes).length;
 
-  if (requiredScopesRemoved || optionalScopesRemoved) {
+  if (!requiredScopesRemoved && !optionalScopesRemoved) {
+    return {
+      operation: CaveatMutatorOperation.Noop,
+    };
+  }
+
+  const updatedCaveatValue = {
+    requiredScopes: Object.fromEntries(newRequiredScopes),
+    optionalScopes: Object.fromEntries(newOptionalScopes),
+  };
+
+  const ethChainIds = getPermittedEthChainIds(updatedCaveatValue);
```

Comment:
I'm having trouble understanding what this function is responsible for, perhaps because of a pre-existing error in the description. It says that it's meant to handle removing an account specifically, but it looks like it's used for removing chains.

Assuming I'm on the right track there, I was wondering, is this meant to handle non-EVM scopes as well? If so, maybe we can determine this in a less EVM specific way (e.g. count the number of scopes, remove if there are zero when excluding `wallet:*`).

Or if it is easier for us to bake in an assumption about these being EVM scopes for now, maybe we can call that out somewhere in a comment